src/Components/Server/src/CircuitDisconnectMiddleware.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    internal class CircuitDisconnectMiddleware
```

Comment:
Is the reason to use a middleware here just so you can use DI? If so, consider adding a comment.

Reply:
Is it though?

I used it for DI, but also because its better to have a middle-ware you can tests than a bunch of code hidden within the extension method.